generator/internal/tracing_connection_generator.cc

```diff
@@ -297,7 +315,37 @@ future<StatusOr<$longrunning_deduced_response_type$>>)""",
   internal::OTelScope scope(span);
   return internal::EndSpan(std::move(span), child_->$method_name$(request));
 }
-)""");
+)""",
+        R"""(
+StatusOr<$longrunning_operation_type$>)""",
+        // TODO(#14344): Remove experimental tag.
+        R"""(
+$tracing_connection_class_name$::$method_name$(google::cloud::ExperimentalTag,
+    google::cloud::NoAwaitTag, $request_type$ const& request) {
+  auto span = internal::MakeSpan(
+      "$product_namespace$::$connection_class_name$::$method_name$");
+  internal::OTelScope scope(span);
```

Comment:
`s/internal::OtelScope/opentelemetry::trace::Scope`

What you have works, it just does some unnecessary extra work (duplicating OTel's thread_local state) that we do not need to do in the synchronous case.